src/Api/Monetization/Structure/Reports/Criteria/AbstractCriteria.php

```diff
@@ -125,7 +119,10 @@ public function getApps(): array
     /**
      * @param string ...$appIds
      *
-     * @return \self
+     * @return self
+     *
+     * @deprecated in 3.0.7, will be removed in 4.0.0. No longer needed.
+     * https://github.com/apigee/apigee-client-php/issues/373
      */
     public function apps(string ...$appIds): self
```

Comment:
I'd call the new method in the deprecated method for DRY.

Reply:
like this inside deprecated methods?

> $this->apps = setApps($appIds);

or

> return setApps($appIds);